imagetest/test_suites/image_boot/image_boot_test.go

```diff
@@ -0,0 +1,50 @@
+// +build linux
+
+package imageboot
+
+import (
+	"flag"
+	"os"
+	"syscall"
+	"testing"
+)
+
+var (
+	runtest = flag.Bool("runtest", false, "really run the test")
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if *runtest {
+		os.Exit(m.Run())
+	} else {
+		os.Exit(0)
+	}
+}
+
+func TestGuestBoot(t *testing.T) {
+	err := syscall.Uname(&syscall.Utsname{})
+
+	if err != nil {
+		t.Fatalf("couldn't get system information, image boot failed")
+	}
+
+}
+
+func TestGuestReboot(t *testing.T) {
+	_, err := os.Stat("/tmp/boot")
+	if os.IsNotExist(err) {
+		// first boot
+		if _, err := os.Create("/tmp/boot"); err != nil {
+			t.Fatal("fail to create file when first boot")
+			return
+		}
+		return
```

Comment:
i think the file not existing is a failure condition - you've left yourself open to passing the test when the file doesn't exist on second boot.

Reply:
I thought If the file doesn't exist on second boot. That implied the file are not created successfully during the first boot.  It must be failed in this line `t.Fatal("fail to create file when first boot")` during the first boot. So I `return`.

Reply:
t.Fatal exits the test, so return not reachable. and you return in the outer loop anyway, so not necessary.

it could not exist for many reasons, so you must fail if the file does not exist. so the first boot will be a fail, the second boot will be a success.

Reply:
Gotcha. Thanks for explanation.